### ISSUE TITLE
Proxy Support broken on Slack App approach (Slack Bot Token approach)

### ISSUE THREAD

**Issue by frednntelia** [slackapi/slack-github-action#217]
### Description

When using the slack-github-action in GitHub App mode, the Action fails to utilize environment variables `HTTPS_PROXY` and `https_proxy` correctly.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/slack-github-action/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Reproducible in:

package version: slackapi/slack-github-action@v1.24.0

node version:

OS version(s): Ubuntu 20.04 LTS

#### Steps to reproduce:

1. run slack-github-action behind a proxy
2. make sure you set HTTPS_PROXY and https_proxy env vars
3. use slack webhook method
4. observe success
5. use slack app method (slack bot token)
6. observe failure

#### Expected result:

HTTPS_PROXY var is respected and slack api call goes through properly, resulting in the display of a message on the respective channel.

#### Actual result:

Slack message never arrives in channel. Likely because HTTPS_PROXY env var is not being respected

#### Attachments:

```
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 3.***.***.153:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 18.***.***.***:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 3.***.***.98:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 3.***.***.98:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 18.***.***.225:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 3.***.***.98:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 18.***.***.***:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 3.***.***.153:443
[WARN]  web-api:WebClient:0 http request failed connect ETIMEDOUT 18.***.***.225:443
Error: The operation was canceled.
```

**Comment by WilliamBergamin** [comment]
Hello @frednntelia thanks for writing in 💯 

This is a curious behavior, would be able to tell me if the proxy you are using is `http` or `https`?

Based on this [test suite](https://github.com/slackapi/slack-github-action/blob/c70e7951e1cb3abd3dbbb4d996d9fa83473a62ff/test/slack-send-test.js#L204) seems like webhook method `should use default axios config when proxy uses HTTPS`

**Comment by frednntelia** [comment]
Hi @WilliamBergamin, thanks for your swift response! 👍🏾 
This proxy in my assembly is http afaik.

**Comment by frednntelia** [comment]
@WilliamBergamin is there a way to enable debug level logs from this Action, so that we can get some more context?

**Comment by WilliamBergamin** [comment]
Based on what I can see in the code there seems to be a bug where HTTP proxies are not properly set but HTTPS proxies are, I am not sure if there is a reason behind this

**Comment by minjunk** [comment]
The Slack Bot's Proxy was updated in June (slackapi/slack-github-action#205), but the release was last in May ([v1.24.0](https://github.com/slackapi/slack-github-action/releases/tag/v1.24.0)).
Do you happen to know when there will be a release plan for this update?

**Comment by joshsnelling** [comment]
Is there a plan to release this fix? If not, is there a workaround that can be used in the meantime?

**Comment by zimeg** [comment]
👋 Just [confirmed](https://github.com/zimeg/slack-sandbox/pull/270) that support for the `proxy` input is still working for both `method` and `webhook` techniques in `slackapi/slack-github-action@v2.0.0`! At least when sending to HTTP proxies 👀 

It seems like support for a proxy with the `token` method was also released in [`v1.2.5`](https://github.com/slackapi/slack-github-action/releases/tag/v1.25.0), but we might've let this issue slip!

The [current documentation](https://github.com/slackapi/slack-github-action?tab=readme-ov-file#proxying-https-requests) gives a bit more info on using this, but for all techniques the `proxy` input can be used:

```yaml
- name: Post to a Slack channel via a proxy
  uses: slackapi/slack-github-action@v2.0.0
  with:
    method: chat.postMessage
    proxy: "http://proxy.example.org:8080" # Change this to a custom value
    token: ${{ secrets.SLACK_BOT_TOKEN }}
    payload: |
      channel: ${{ secrets.SLACK_CHANNEL_ID }}
      text: "This message was sent through a proxy"
```

Going to close this issue as completed, but please feel free to follow up with questions if this doesn't seem to be working as expected 🙏
